addons/binding/org.openhab.binding.isy/src/main/java/org/openhab/binding/isy/handler/IsyHandlerFactory.java

```diff
@@ -118,7 +119,7 @@ protected ThingHandler createHandler(Thing thing) {
                     .addChannelforDeviceId(CHANNEL_SMOKEDETECT_LOWBAT, 6)
                     .addChannelforDeviceId(CHANNEL_SMOKEDETECT_MALFUNCTION, 7).build();
         } else if (thingTypeUID.equals(VENSTAR_THERMOSTAT_THING_TYPE)) {
-            return IsyHandlerBuilder.builder(thing).addChannelforDeviceId(CHANNEL_VENSTAR_MAIN, 1)
+            return VenstarThermostatHandlerBuilder.builder(thing).addChannelforDeviceId(CHANNEL_VENSTAR_MAIN, 1)
```

Comment:
Thomas.  I may have confused you with my suggestion before.  I don't think we need a builder for the Thermo handler.  I think within the handler factory you should just instantiate it directly.

eg. return new VenstarThemostateHandler(thing)

I just used that builder pattern because for a lot of the isy devices, it is really just handling on/off commands, so I went the route of creating generic code.  For the thermostat, I think you need a custom handler as you have written.

Here is an example of the factory for another binding I worked on with one of the more prolific committers for Openhab.  Check out how we just directly instantiate the handlers based on the thing type.  this would eliminate your thermo builder class, and all that channel mapping which is done for the more generic isy handlers would just be encapsulated within the thermostat handler.

https://github.com/QuailAutomation/openhab2-addons/blob/digitaldan-omnilink-binding-thing-types/addons/binding/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/OmnilinkHandlerFactory.java

Let me know what you think.

Reply:
Well, the problem is that the Venstar is a mix between the two - one channel is special, the remaining 3 are handled like any other you already implemented. One reason I followed your pattern is that I don't know if any other Thermos use the same channel id's  for channels handled the standard way  (which then could be adapted in that factory for that new device) Alternatively, we could just create another class for any new thermo device.

But I guess you're right, we could worry about that when that actually comes up. I'll make a change to just set everything up in the handler ctor for now..

Reply:
ok, 3rd is a charm (I hope :) )
Commit 7a8f19413a28eade52ee3792f8b666f7a825ec34